department-of-veterans-affairs/caseflow
# Kamalam7/appeals 34927

Resolves (https://jira.devops.va.gov/browse/APPEALS-34927)

# Description
If the Package document type is 10182, it will show the Intake Appeal button. When we Click on Intake Appeal button, in the backend it will assign the Review package task to the current user and cancelled. The tasks associated to the Correspondence is cancelled.  
## Acceptance Criteria
- [x] When the mail user clicks the "Intake Appeal" button the following happens to the Correspondence:
       Its Review Package Task instructions are updated to say "An appeal intake was started because this 
       Correspondence is a 10182"
       It is removed from Caseflow
       Its Review Package Task is assigned to the current mail user and canceled
       Cancel the tasks associated with Correspondence 

## Testing Plan
1. Login as MEAWADBVAJ.
2.  In the address bar go to /queue/correspondence
3. Click on any correspondence with 10182 package document type.
4. This will update the Review package task instructions and assigned to the current user and cancelled the Review package task and its Correspondence in the Tasks table.
5. In the rails console, Create the Correspondence with this following code
          Correspondence.create!(
          uuid: SecureRandom.uuid,
          portal_entry_date: Time.zone.now,
          source_type: "Mail",
          package_document_type_id: 15,
          correspondence_type_id: 8,
          cmp_queue_id: 1,
          cmp_packet_number: 9999999999,
          va_date_of_receipt: Time.zone.yesterday,
          notes: "This is a note from CMP.",
          assigned_by_id: 81,
          veteran_id: 1
        )
6. In the database check the tasks table and in the appeal_type as Correspondence.
7. Now pickup the uuid from the correspondence you created. 
8. In the Review package page address bar change the uuid from the Correspondence you created, http://localhost:3000/queue/correspondence/dc981f56-a1c0-4fdf-98ac-0be6afa7222a/review_package
9. Now in Caseflow, in the Review package page click on the Intake appeal button.
10. This will update the Review package task instructions and assigned to the current user and cancelled the Review package task and its Correspondence in the Tasks table.



# Frontend
## User Facing Changes

 BEFORE|AFTER
 ---|---

## Storybook Story
*For Frontend (Presentation) Components*

# Backend
## Database Changes
*Only for Schema Changes*


## Integrations: Adding endpoints for external APIs

# Best practices
## Code Documentation Updates

## Tests
### Test Coverage
Did you include any test coverage for your code? Check below:

### Code Climate
Your code does not add any new code climate offenses? If so why?

## Monitoring, Logging, Auditing, Error, and Exception Handling Checklist

### Monitoring

### Logging

### Auditing

### Error Handling

### Exception Handling
